docs/gallery/torontonian_tutorial.ipynb

```diff
@@ -0,0 +1,350 @@
+    "where $\\text{P}(|l|)$ stands for the power set of $\\{0,...,l-1\\}$.\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We could likewise define it as a proof-of-concept with simple Python code:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 222,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def powerset(s): #empty set handled as special case\n",
+    "  import itertools\n",
+    "  for i in range(1, len(s)+1):\n",
+    "    yield from itertools.combinations(s, i)\n",
+    "#https://the-walrus.readthedocs.io/en/latest/code/quantum.html pip install thewalrus\n",
+    "def matix(mat, ix): return mat[ix][:,ix]\n",
+    "def torontonian(mat):\n",
+    "  N = len(mat); assert((N & 1) == 0) #N=2*d\n",
+    "  N >>= 1 #determinant of empty matrix is 1\n",
+    "  return sum(\n",
+    "    (1 if ((N - len(Z)) & 1) == 0 else -1)\n",
+    "     / np.sqrt(np.linalg.det(np.eye(len(Z)<<1)\n",
+    "               - matix(mat, [*Z, *(x + N for x in Z)])))\n",
+    "     for Z in powerset(list(range(N)))) + (1 if (N & 1) == 0 else -1)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Using the library"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Import the library in the usual way:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 204,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from thewalrus import tor"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To use it  we need to pass square numpy arrays as arguments, thus we also must import NumPy:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 205,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The library provides functions to compute Torontonians of real and complex matrices. The functions arguments must be passed as the NumPy arrays or matrices."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 226,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "((-16-16j), (-16-16j))"
+      ]
+     },
+     "execution_count": 226,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "size = 10\n",
+    "matrix = 1j*np.ones([size,size])\n",
+    "matrix = np.tril(matrix.conjugate(), -1) + np.triu(matrix, 1) + np.eye(size)\n",
+    "torontonian(matrix), tor(matrix)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Not surprisingly, the Torontonian of a matrix containing only ones on the main diagonal, and its upper triangular is $1i$ while its lower triangular is $-1i$ is given by $-1^l(2^{l-1}+2^{l-1}i)$"
```

Comment:
This sentence reads a bit strange. I don't really understand what it's trying to say. 🤔

Reply:
Just giving a sort of most basic style computation.  Its a little strange indeed, I basically just took everything from the Hafnian tutorial.

Reply:
Corrected.